### PR TITLE
Made docs adhere to style guide

### PR DESCRIPTION
**Pull request by jc-berger**
Since there were merge conflicts with #39, I've opened this PR instead. 

This PR addresses the doc and language concerns while avoiding a merge conflict.

### CODE REVIEW

File: docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc [hunk 1]
```diff
@@ -24,10 +24,11 @@ This guide explains how to migrate an existing v1.0 devfile to v2.0.
 [role="_additional-resources"]
 .Additional resources
 
-* New metadata: `version`. See link:https://github.com/che-incubator/devworkspace-api/issues/10[corresponding issue].
-* New mandatory metadata for plugins. See link:https://github.com/devfile/api/issues/31[corresponding issue].
+For a deeper understand of migrating devfiles, go to the following GitHub issues:
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc:27]: I think you meant "deeper understanding". ;)

Honestly, I'd also say something like "for more information". We can't guarantee that someone will get a deeper understanding, but we can certainly link them to more details!

File: docs/modules/user-guide/partials/proc_adding-event-bindings.adoc [hunk 2]
```diff
@@ -2,11 +2,7 @@
 = Adding event bindings
 
 [role="_abstract"]
-Commands can be bound to events to form four different types of events: `preStart`, `postStart`, `preStop`, and `postStop`.
-
-.Prerequisites
-
-
+This section describes how to add an event to devfiles v2.x. An event specified in a devfile v1.x still works in a devfile v2.x.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:5]: The plural "devfiles v2.x" reads very oddly to me. Should we change that to "v2.x devfiles", or has this been discussed elsewhere?

File: docs/modules/user-guide/partials/proc_adding-event-bindings.adoc [hunk 3]
```diff
@@ -46,4 +42,5 @@ events:
 
 [role="_additional-resources"]
 .Additional resources
-* See https://github.com/che-incubator/devworkspace-api/issues/32[corresponding issue].
+
+For a deeper understand of adding event bindings, go to the following GitHub issue: link:https://github.com/devfile/api/issues/32[lifecycle bindings to bind commands].
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:46]: Another "understand". :)

File: docs/modules/user-guide/partials/proc_migrating-components.adoc [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 [role="_abstract"]
-This section describes how to migrate components to devfile v2. There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
+This section describes how to migrate components to devfiles v2.x. A component specified in a devfile v1.x no longer works in a devfile v2.x
 
 .Procedure
 
 . Component is a polymorphic type
 +
-For a better syntax validation component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
+For a better syntax, the validation component is now defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_migrating-components.adoc:11]: Not sure what "for a better syntax" adds to this (and the article feels unnecessary in any case).

File: docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc [hunk 5]
```diff
@@ -2,20 +2,20 @@
 = Migrating plug-ins
 
 [role="_abstract"]
-This section describes how to migrate plug-ins to devfile v2.x. Plug-ins are now specified using a devfile. A plug-in specified in a v1.0 `meta.yaml` does not work as it is in a devfile v2.x.
+This section describes how to migrate plug-ins to devfile v2.x. A plug-in specified in a meta.yaml in a devfile v1.x no longer works in a devfile v2.x.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:5]: Looks like `meta.yaml` needs to be formatted.

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc [hunk 6]
```diff
@@ -2,17 +2,22 @@
 = Refering to a parent devfile in a devfile
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:2]: Not part of this PR, but wanted to note that "referring" is spelled incorrectly. :)

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc [hunk 7]
```diff
@@ -2,17 +2,22 @@
...
 
 [role="_abstract"]
-It is possible to refer to another devfile as a parent to a given devfile. If a parent is specified, the devfile will inherit all behaviour from the parent. Part of the content of the parent can be overridden in the child devfile. That allows refering a devfile (the parent) from a distinct devfile and makes it possible to reuse in multiple devfiles the same parent (the stack).
+This section describes how to designate a parent devfile to a given devifle. If you designate a parent devfile, the given devfile inherits all its behavior from its parent.  Still, you can use the child devfile to override certain content from the parent devfile. If you override the correct content, you can reuse the same parent devfile in multiple other devfiles. If you do reuse a parent devfile, the parent turns into a stack that is used in multiple other devfiles.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:5]: Typo: "devifle"

I actually think the original opening sentence is a decent introduction before going into more detail. "This section describes..." feels superfluous.

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc [hunk 8]
```diff
@@ -2,17 +2,22 @@
...
+* Define the schemaVersion to v2.x. If you need to define your schemaVersion, go to xref:adding-schema-version-to-a-devfile.adoc[].
+* Define a name for the devfile. If you need to define a name for your devfile go to xref:adding-a-name-to-a-devfile.adoc[].
 
 
 .Procedure
 
-* A parent devfile can be referenced in 3 different ways: `registry`, `uri` or `kubernetes`.
+You can refer to a parent devfile in three different ways:
+
+* registry
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:17]: Shouldn't these elements be formatted like they were before?

File: docs/modules/user-guide/partials/proc_adding-event-bindings.adoc (hunk 2 above)
Comment (johnmcollier) [reply docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:5]: +1 for v2.x devfiles, sounds more natural to me too

File: docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc (hunk 1 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc:27]: good point. I addressed this concern in all the docs. resolving this conversation now!

File: docs/modules/user-guide/partials/proc_adding-event-bindings.adoc (hunk 2 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:5]: Awesome. I've changed the order to `v2.x devfiles` and `v1.x devfiles` in all the docs. Resolving this conversation as I've addressed these concerns.

File: docs/modules/user-guide/partials/proc_migrating-components.adoc (hunk 4 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_migrating-components.adoc:11]: Eliminated "for a better syntax" I kept "_the_ validation component..." do you think it reads fine?

File: docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc (hunk 5 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:5]: Done!

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc (hunk 7 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:5]: I think we wanted to start each abstract with "this section describes..." at least until we have a more user centric abstract written. 

Do you think it's fine as is? Or should I change the first sentence for all 8 docs?

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc (hunk 8 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:17]: Eva said it's good to give context for the prereqs and I thought it did help users know if they should even bother clicking the links. Do you think the sentence, "define the...if you need to define..." is unhelpful? Or can it be kept?

**Comment by jc-berger** [comment]
New commit, "more doc changes" is up, thanks!

**Review by johnmcollier — APPROVED** [review]
LGTM from dev

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc (hunk 8 above)
Comment (robin-owen) [reply docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:17]: I just meant that elements such as `registry` should use the same formatting (monospace). :)

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc (hunk 7 above)
Comment (robin-owen) [reply docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc]: It's probably fine as-is for now, but I do think we should come back later and reevaluate how we're writing each abstract (if that makes sense).

Primarily, I'm concerned with the fact that this feels like fluff. Describing what a section is about to say is unnecessary if it's written properly in the first place, and I feel like the original first sentence of this section did so decently well. Why change it in that case? You're working against the future work across all of the abstracts.

File: docs/modules/user-guide/partials/proc_migrating-components.adoc (hunk 4 above)
Comment (robin-owen) [reply docs/modules/user-guide/partials/proc_migrating-components.adoc]: Reads well to me. :)

File: docs/modules/user-guide/partials/proc_migrating-components.adoc [hunk 9]
```diff
@@ -2,13 +2,13 @@
...
 [role="_abstract"]
-This section describes how to migrate components to devfile v2. There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
+This section describes how to migrate components to v2.x devfiles. A component specified in a v1.x devfile no longer works in a v2.x devfile.
 
 .Procedure
 
 . Component is a polymorphic type
 +
-For a better syntax validation component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
+The validation component is now defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_migrating-components.adoc:11]: We should use a serial comma here (when listing types at the end of the sentence) according to the IBM style guide.

File: docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc [hunk 10]
```diff
@@ -11,7 +11,7 @@ endif::[]
 
 :context: assembly_migrating-to-devfile-v2
 
-This guide explains how to migrate an existing v1.0 devfile to v2.0.
+The following documents explain how to migrate an existing v1.x devfile to a v2.x devfile:
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc:14]: Is there a reason this was changed from "this guide" to "the following documents"?

File: docs/modules/user-guide/partials/proc_migrating-components.adoc (hunk 4 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_migrating-components.adoc]: resolving this conversation then!

File: docs/modules/user-guide/partials/proc_migrating-projects.adoc [hunk 11]
```diff
@@ -2,26 +2,25 @@
...
 
 [role="_abstract"]
-There are NO major changes in the `projects` section of the devfile. A project specified in a v1.0 devfile will work as it is in a devfile v2.0.
+This section describes how to migrate projects to v2.x devfiles. A project specified in a v1.x devfile still works in a v2.x devfile.
 
 .Procedure
 
-*  The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user pick one) and the git repo get not be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
-+
+V2.x devfiles has a new kind of project: `starterProjects`. Pick only one `starterProject` and use it in interactive mode. With a `starterProject`, you only need to copy your source code, so you no longer need to clone your git repository.
```
[the remark this answers is not in the document]
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_migrating-projects.adoc:9]: Thanks, Kevin. I also didn't like starting the sentence with `V2.x...` because the capitalized `v` looked awkward to me as well. Great revision which I implemented!

File: docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc (hunk 10 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc:14]: Yes, I thought "documents" made more sense because Eva pointed out that this document is not a guide; hence, it's not very accurate for us to say 'guide' in this instance.

File: docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc (hunk 7 above)
Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc]: Cool, let's address this concern in #402: https://github.com/devfile/api/issues/402

File: docs/modules/user-guide/partials/proc_migrating-components.adoc [hunk 12]
```diff
@@ -2,13 +2,13 @@
...
 [role="_abstract"]
-This section describes how to migrate components to devfile v2. There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
+This section describes how to migrate components to v2.x devfiles. A component specified in a v1.x devfile no longer works in a v2.x devfile.
 
 .Procedure
 
 . Component is a polymorphic type
 +
-For a better syntax validation component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
+The validation component is now defined as a polymorphic type and can be implemented as, `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_migrating-components.adoc:11]: Misplaced comma:

```adoc
...and can be implemented as `container`, `kubernetes`, `openshift`, `plugin`, or `volume`.
```

File: docs/modules/user-guide/partials/proc_migrating-projects.adoc [hunk 13]
```diff
@@ -2,26 +2,25 @@
...
 
 [role="_abstract"]
-There are NO major changes in the `projects` section of the devfile. A project specified in a v1.0 devfile will work as it is in a devfile v2.0.
+This section describes how to migrate projects to v2.x devfiles. A project specified in a v1.x devfile still works in a v2.x devfile.
 
 .Procedure
 
-*  The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user pick one) and the git repo get not be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
-+
+A new `starterProjects` project type can be defined in v2.x devfiles. Pick only one `starterProject` and use it in interactive mode. With a `starterProject`, you only need to copy your source code, so you no longer need to clone your Git repository.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_migrating-projects.adoc:9]: The first `starterProjects` should be `starterProject`.

File: docs/modules/user-guide/partials/proc_adding-event-bindings.adoc [hunk 14]
```diff
@@ -2,11 +2,7 @@
 = Adding event bindings
 
 [role="_abstract"]
-Commands can be bound to events to form four different types of events: `preStart`, `postStart`, `preStop`, and `postStop`.
-
-.Prerequisites
-
-
+This section describes how to add an event to v2.x devfiles. An event specified in a v1.x devfile still works in a v2.x devfile.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:5]: Singular/plural mismatch.

```adoc
This section describes how to add events to v2.x devfiles.
```

Or:

```adoc
This section describes how to add an event to a v2.x devfile.
```

File: docs/modules/user-guide/partials/proc_migrating-commands.adoc [hunk 15]
```diff
@@ -2,7 +2,7 @@
 = Migrating commands
 
 [role="_abstract"]
-This section describes how to migrate the `commands` section to devfile v2.x. A command specified in a v1.0 devfile does not work as it is in a devfile v2.x.
+This section describes how to migrate the `commands` section to v2.x devfiles. A command specified in a v1.x devfile does not work in a v2.x devfile.
```
Comment (robin-owen) [inline docs/modules/user-guide/partials/proc_migrating-commands.adoc:5]: Singular/plural mismatch.

```adoc
This section describes how to migrate the `commands` section to a v2.x devfile.
```

**Review by robin-owen — APPROVED** [review]
LGTM. :+1:

**Review by rkratky — APPROVED** [review]
